packages/flutter_tools/test/general.shard/plugins_test.dart

```diff
@@ -1073,6 +1130,112 @@ flutter:
+    ''');
+
+        final FlutterManifest manifest = FlutterManifest.createFromString('''
+name: test
+version: 1.0.0
+
+dependencies:
+  url_launcher: ^1.0.0
+  user_selected_url_launcher_implementation: ^1.0.0
+    ''', logger: BufferLogger.test())!;
+
+        flutterProject.manifest = manifest;
+
+        await injectPlugins(flutterProject, linuxPlatform: true);
+
+        final File registrantImpl = linuxProject.managedDirectory.childFile('generated_plugin_registrant.cc');
+
+        expect(registrantImpl.existsSync(), isTrue);
+        expect(registrantImpl.readAsStringSync(), contains('user_selected_url_launcher_linux_register_with_registrar'));
+        expect(registrantImpl.readAsStringSync(), isNot(contains('inline_url_launcher_linux_register_with_registrar')));
+      }, overrides: <Type, Generator>{
+        FileSystem: () => fs,
+        ProcessManager: () => FakeProcessManager.any(),
+      });
+
+      testUsingContext('Injects user selected implementation despite default implementation', () async {
+        final List<Directory> directories = createFakePlugins(fs, <String>[
+          'user_selected_url_launcher_implementation',
+          'url_launcher',
+          'url_launcher_linux',
+        ]);
+
+        // Add inline native implementation to `user_selected_url_launcher_implementation`
+        directories[0].childFile('pubspec.yaml').writeAsStringSync('''
+flutter:
+  plugin:
+    implements: url_launcher
+    platforms:
+      linux:
+        pluginClass: UserSelectedUrlLauncherLinux
+    ''');
+
+        // Add default native implementation to `url_launcher`
+        directories[1].childFile('pubspec.yaml').writeAsStringSync('''
+flutter:
+  plugin:
+    platforms:
+      linux:
+        default_package: url_launcher_linux
+    ''');
+
+        // Add inline native implementation to `url_launcher_linux`
+        directories[1].childFile('pubspec.yaml').writeAsStringSync('''
+flutter:
+  plugin:
+    platforms:
+      linux:
+        pluginClass: InlineUrlLauncherLinux
+    ''');
+
+        final FlutterManifest manifest = FlutterManifest.createFromString('''
+name: test
+version: 1.0.0
+
+dependencies:
+  url_launcher: ^1.0.0
+  user_selected_url_launcher_implementation: ^1.0.0
+    ''', logger: BufferLogger.test())!;
+
+        flutterProject.manifest = manifest;
+
+        await injectPlugins(flutterProject, linuxPlatform: true);
+
+        final File registrantImpl = linuxProject.managedDirectory.childFile('generated_plugin_registrant.cc');
+
+        expect(registrantImpl.existsSync(), isTrue);
+        expect(registrantImpl.readAsStringSync(), contains('user_selected_url_launcher_linux_register_with_registrar'));
+        expect(registrantImpl.readAsStringSync(), isNot(contains('inline_url_launcher_linux_register_with_registrar')));
+      }, overrides: <Type, Generator>{
+        FileSystem: () => fs,
+        ProcessManager: () => FakeProcessManager.any(),
+      });
+
```

Comment:
There should also be a "user-selected implementation overrides default implementation" test.

Reply:
The test for it can be found [here](https://github.com/flutter/flutter/pull/137040/files/37ce21a454de466e589ff96cd39684a46c4f4029#diff-4b59d4ff0ace84a3aaa79567f0588a1bb72a8cfb72558a9e1548ff33b0de8ffbR1181).

I suspect you meant a test for `Build time plugin injection` aka **web**. I added a test for it to ensure the injection code is also working for web in `injectBuildTimePluginFiles` in `flutter_plugins.dart`. The plugin resolution code is shared with the native code, which is already covers this case. But if it feels more safe, I can add this test, too (?)